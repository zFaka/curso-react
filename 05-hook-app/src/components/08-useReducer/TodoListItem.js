import React from 'react';

const TodoListItem = ({todo, index, handleDelete, handleToggle}) => {
  return (
      <>
                <li
                  key={todo.id}
                  className='list-group-item'
                >
                  <p 
                    className={`${todo.done && 'complete'}`}
                    onClick={()=>handleToggle(todo.id)}
                  >
                    {++index}&emsp;{todo.desc}
                  </p>

                  <button 
                    className='btn btn-danger'
                    onClick={()=>handleDelete(todo.id)}
                  >
                    x 
                  </button>
                </li>
      </>
  )}

export default TodoListItem;
