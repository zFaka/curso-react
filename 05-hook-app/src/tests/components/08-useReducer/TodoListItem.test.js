import {shallow} from "enzyme";
import TodoListItem from "../../../components/08-useReducer/TodoListItem";
import {demoTodos} from "../../fixtures/demoTodos";

describe('Test TodoListItem', () => {

  const handleToggle = jest.fn();
  const handleDelete = jest.fn();

  const wrapper = shallow(
    <TodoListItem
      todo={demoTodos[0]} 
      index={0}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );




  test('Should render', () => {
    expect(wrapper)
      .toMatchSnapshot()
  })




  test('Should call handleDelete', () => {


    wrapper.find('button').simulate('click'); 



    expect(handleDelete)
      .toHaveBeenCalledWith(demoTodos[0].id);
  })




  test('Should call handleToggle', () => {


    wrapper.find('p').simulate('click'); 



    expect(handleToggle)
      .toHaveBeenCalledWith(demoTodos[0].id);
  })




  test('Should show output text', () => {


    const textInsideP = wrapper.find('p');



    expect(textInsideP.text().trim())
      .toBe(`${demoTodos[0].desc}`);
  })




  test('Should show complete if todo is done', () => {

    const todo = demoTodos[0];
    todo.done = true

    const wrapper = shallow(
      <TodoListItem
        todo={todo} 
      />
    );
    expect(wrapper.find('p').hasClass('complete'))
      .toBe(true);
  })
})
