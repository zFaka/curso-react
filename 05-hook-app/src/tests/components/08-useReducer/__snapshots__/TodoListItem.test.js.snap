// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TodoListItem Should render 1`] = `
<Fragment>
  <li
    className="list-group-item"
    key="1"
  >
    <p
      className="false"
      onClick={[Function]}
    >
      lrn rct
    </p>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      x
    </button>
  </li>
</Fragment>
`;
